src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -97,14 +98,17 @@ internal sealed class SolutionRestoreWorker : SolutionEventsListener, ISolutionR
             Lazy<Common.ILogger> logger,
             Lazy<INuGetErrorList> errorList,
             Lazy<IOutputConsoleProvider> outputConsoleProvider,
-            Lazy<INuGetFeatureFlagService> nugetFeatureFlagService)
+            Lazy<INuGetFeatureFlagService> nugetFeatureFlagService,
+            [Import(typeof(IInfoBarService))]
```

Comment:
I don't think you need this. There shouldn't be multiple `IInfoBarService` exported.